pkg/operation/controlplanesecrets/secrets_deployer_test.go

```diff
@@ -0,0 +1,221 @@
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		k8sClient = mockclient.NewMockClient(ctrl)
+
+		gardenerResourceDataList = gardencorev1alpha1helper.GardenerResourceDataList{
+			v1alpha1.GardenerResourceData{
+				Name: common.BasicAuthSecretName,
+				Type: string(secretsinfodata.BasicAuthDataType),
+				Data: runtime.RawExtension{Raw: []byte("{\"password\":\"asdf\"}")},
+			},
+			v1alpha1.GardenerResourceData{
+				Name: common.StaticTokenSecretName,
+				Type: string(secretsinfodata.StaticTokenDataType),
+				Data: runtime.RawExtension{Raw: []byte("{\"tokens\":[{\"userName\":\"foo\",\"userID\":\"fooID\",\"groups\":[\"group1\",\"group2\"],\"token\":\"foo\"}]}")},
+			},
+			v1alpha1.GardenerResourceData{
+				Name: "ca",
+				Type: string(secretsinfodata.CertificateDataType),
+				Data: runtime.RawExtension{Raw: []byte(fmt.Sprintf("{\"privateKey\":\"%s\",\"certificate\":\"%s\"}", []byte(cakey), []byte(cacert)))},
```

Comment:
Nit - I guess the `[]byte` casting here is unnecessary:

```suggestion
				Data: runtime.RawExtension{Raw: []byte(fmt.Sprintf("{\"privateKey\":\"%s\",\"certificate\":\"%s\"}", cakey, cacert))},
```